SNT/NativeMemoryManager.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.Buffers;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System;
+
+namespace Torch.SNT
+{
+    /// <summary>
+    ///   Class used to wrap the native memory of a Torch Tensor.  
+    /// </summary>
+    internal class NativeMemory<T> : MemoryManager<T>
+    {
+        private int refCount = 0;
+        private IntPtr memory;
+        private int length;
+
+        public NativeMemory(IntPtr memory, int length)
```

Comment:
This probably should do some defensive parameter checks of `memory` not being null and such, and `length` being greater than 0.

Reply:
Length could be zero if you want to set up an empty tensors I presume. I can add the check that length is not negative and memory is not null.